### ISSUE TITLE
DNS: Additional and wrong CNAME added by "System Status Checks"

### ISSUE THREAD

**Issue by pm-pm** [mail-in-a-box/mailinabox#1686]
Hello,

i have to use **external DNS**. **All entrys are set as recommended** by "Mail in a Box - External DNS - Page". External DNS is ok and well working.

My personal Sub-Domain: `cherry.otherdomain.com` - **My Server IP**: 111.111.111.111
Domain of company: `otherdomain.com` - Company Domain IP:  222.222.222.222

Here the problem:

After startup my MIAB Server:
`nslookup cherry.otherdomain.com` --> 111.111.111.111 - OK
1 hour later (time does not matter)
`nslookup cherry.otherdomain.com` --> 111.111.111.111 - Still OK
Using "MIAB System Status Checks Page"
`nslookup cherry.otherdomain.com` --> `CNAME to otherdomain.com` --> 222.222.222.222 - ERROR.

So the "MIAB System Status Checks Page Script" adds a CNAME for my 3rd level Domain and breaks my installation. I have not found the point in the sourcecode the stop this. :-(

Using "rndc flush" corrects the situation and removes the CNAME, but after using "MIAB System Status Checks Page Script" for the next time the WRONG CANME entry is there again. :-(

Please remove this wrong CNAME entry when using "MIAB System Status Checks Page Script" or tell me where to remove it in the script code. After two nights of searching i need your help.

Thanks

pm-pm

**Comment by jvolkenant** [comment]
> i have to use **external DNS**. **All entrys are set as recommended** by "Mail in a Box - External DNS - Page". External DNS is ok and well working.
> 

If you have nameservers set to an external service and not your MIAB box.  Your external name service needs to be updated with your dns entries.

It is a little confusing as to what is wrong.

**Comment by pm-pm** [comment]
Yes you are right. Nameserver are set to external Service.
All entrys are set as recommended in "MIAB External DNS Page" !! :-)
There is no CNAME recommended!
So the DNS configuration is correkt.

On e.g. my Windows box i can see all DNS as correct.

BUT when you login in with ssh to your MIAB Box and use e.g. nslookup (after using "MIAB System Status Checks Page Script") there is an additional wrong CNAME. 

This CNAME is **only** visible on the MIAB Box. So the Software on the MIAB Software must set this CNAME. 


Why does the "MIAB System Status Checks Page Script" set this additional CNAME ??

**Comment by pm-pm** [comment]
Hello,
i deleted the box and installed it from scratch. Same problem!
External DNS entry at my dns provider looks very fine. Checks from anywhere except the box have not resulted in any problem.

The box adds a wrong CNAME when you use the "MIAB System Status Checks Page Script".

**Comment by pm-pm** [comment]
Has anyone already successfully put a 3rd level domain with external DNS into operation?

Note: 3rd.level.IP is not equal to 2nd.level.IP.

**Comment by victorlap** [comment]
Hi @pm-pm.

I have run a box with external dns without problem. Are you sure your nameservers have been setup correct?

**Comment by pm-pm** [comment]
Yes. :-) I am very sure that everything is ok. After many days of research i can not find any problem here. I am happy about every hint.

A question: Do you use a third level Domain like: "`box.anydomain.com`"?
If yes, Are the IPs "`box.anydomain.com`" and "`anydomain.com`" different or the same?

If the IPs "`box.anydomain.com`" and "`anydomain.com`" are the same, the additional CNAME does not harm. The problem only exist if "`box.anydomain.com`" and "`anydomain.com`" have different IPs.
